docs/source/examples/example_dates.rst

```diff
@@ -0,0 +1,158 @@
+Example: Dates
+==============
+
+This example concerns itself with expressing ``Constraint`` s against data revolving
+around dates. While date ``Constraint`` s between tables exist, we will only illustrate
+``Constraint`` s on a single table and reference values here. As a consequence, we will
+only use ``WithinRequirement``, as opposed to ``BetweenRequirement``.
+
+Concretely, we will assume a table containing prices for a given product.
```

Comment:
To make it more realistic, I'd say it's a products table and add a product id to each row. In this example, all the ids will be the same, but it's clearer even if it's not adding any information.

Reply:
Great idea!